tokio/tests/task_local.rs

```diff
@@ -116,3 +116,64 @@ async fn task_local_available_on_completion_drop() {
     assert_eq!(rx.await.unwrap(), 42);
     h.await.unwrap();
 }
+
```

Comment:
Please add the following tests:

 * Can get id from `spawn_blocking` task.
 * Can get id from `spawn_local` task on a `LocalSet`.
 * Can't get id in `block_on` call.
 * Can get id in future destructor when it completes normally.
 * Can get id in future destructor when it is aborted with `JoinHandle::abort`.
 * Can get id in destructor of output when `JoinHandle` dropped before task completion.
 * Can get id in destructor of output when `JoinHandle` dropped after task completion.
 * If you use a `LocalSet` to run a local task within another task, the original task's id is not erased once we return.
 * If you use `block_in_place` + `spawn` + `block_on` on a current-thread runtime to execute a spawned task within another task, the original task's id is not erased.

Reply:
> Please add the following tests:
> 
> * ~~Can get id from `spawn_blocking` task.~~
> * ~~Can get id from `spawn_local` task on a `LocalSet`.~~
> * ~~Can't get id in `block_on` call.~~
> * ~~Can get id in future destructor when it completes normally.~~
> * ~~Can get id in future destructor when it is aborted with `JoinHandle::abort`.~~
> * Can get id in destructor of output when `JoinHandle` dropped before task completion.
> * Can get id in destructor of output when `JoinHandle` dropped after task completion.
> * If you use a `LocalSet` to run a local task within another task, the original task's id is not erased once we return.
> * If you use `block_in_place` + `spawn` + `block_on` on a current-thread runtime to execute a spawned task within another task, the original task's id is not erased.

Reply:
> Please add the following tests:
> 
> ~~* Can get id from `spawn_blocking` task.~~
> ~~* Can get id from `spawn_local` task on a `LocalSet`.~~
> ~~* Can't get id in `block_on` call.~~
> ~~* Can get id in future destructor when it completes normally.~~
> ~~* Can get id in future destructor when it is aborted with `JoinHandle::abort`.~~
> ~~* Can get id in destructor of output when `JoinHandle` dropped before task completion.~~
> * Can get id in destructor of output when `JoinHandle` dropped after task completion.
> ~~* If you use a `LocalSet` to run a local task within another task, the original task's id is not erased once we return.~~
> * If you use `block_in_place` + `spawn` + `block_on` on a current-thread runtime to execute a spawned task within another task, the original task's id is not erased.

@Darksonn I've added the tests that I've struck through. Can you please check if they make sense? I'm not confident about some of them. I'll add the remaining tests soon.

Reply:
added the last test as well.